### PR TITLE
96358 Fix for out of order attachment ids - IVC CHAMPVA forms

### PR DESCRIPTION
**Pull request by michaelclement**
## Summary

This PR fixes an issue with how attachment metadata is prepared before being sent to Pega for ingestion. Previously, the document types (`attachment_id`, e.g.: 'Birth certificate') were rolled into a list based solely on the order of the front end configuration.

These values were then matched up with the list of attachments in the cache on the backend. However, if a user uploaded files in any order other than the order in which they encounter them, the alignment between attachment_id and the file itself would be thrown off. This caused files to be mis-categorized.

By reorganizing the `attachment_id` array to match with the order files were uploaded, metadata gets matched correctly regardless of how the user uploaded their files.

- Matches supporting docs up to database records and sorts by date created

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/96358

## Testing done

- [X] *New code is covered by unit tests*

## Screenshots
NA

## What areas of the site does it impact?
IVC CHAMPVA forms

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb
Comment (michaelclement) [inline modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb:107]: The order of the attachment IDs has always mattered, so this is ensuring we have thins in a sane order (date created) rather than the previous arbitrary order. 

In future we will probably want to look into a wider refactor. A direct mapping of files to doc types rather than relying on the order of this array would be better, but this is a necessary first step to make sure the MVP works.

File: modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb
Comment (michaelclement) [inline modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb:107]: The order of the attachment IDs has always mattered, so this is ensuring we have things in a sane order (date created) rather than the previous arbitrary order.

In future we will probably want to look into a wider refactor. A direct mapping of files to doc types rather than relying on the order of this array would be better, but this is a necessary first step to make sure the MVP works.

File: modules/ivc_champva/spec/requests/ivc_champva/v1/forms/uploads_spec.rb
Comment (michaelclement) [inline modules/ivc_champva/spec/requests/ivc_champva/v1/forms/uploads_spec.rb:40]: Mock added to account for new logic in `supporting_document_ids` that attempts to read `PersistentAttachments::MilitaryRecords`.

**Review by reiting — DISMISSED** [review]

File: modules/ivc_champva/spec/requests/ivc_champva/v1/forms/uploads_spec.rb
Comment (acrollet) [inline modules/ivc_champva/spec/requests/ivc_champva/v1/forms/uploads_spec.rb:157]: thanks for adding the test!

File: modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb
Comment (acrollet) [inline modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb:95]: OOC what would happen if this query failed? Any possibility of a silent failure? I don't want to hold this PR up but it may be worth a follow-on to investigate and make it more defensive if necessary.

**Review by acrollet — APPROVED** [review]
nice work! just one question in a comment

**Review by cloudmagic80 — APPROVED** [review]

File: modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb
Comment (michaelclement) [reply modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb:95]: If the query returns `nil` on l95, we'll get an error when we try to pull the `created_at` and `file.id` on lines 98 & 99. This will bubble back up to the user as a 500, so they'll get an error screen and the submission won't go through. 

I can definitely make a follow up to probe this a little more - thanks for looking this over!

Comment (acrollet) [reply modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb:95]: thanks @michaelclement! maybe the follow-up could be to make a quick couple of request specs that confirm an error is returned when the query returns `nil` or throws.

Comment (michaelclement) [reply modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb:95]: Definitely a good idea - I'm going to put a couple more tests into this PR while it's top of mind. Thanks!

**Review by rjohnson2011 — APPROVED** [review]

File: modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb
Comment (michaelclement) [reply modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb:95]: @acrollet Tests added in [this commit](https://github.com/department-of-veterans-affairs/vets-api/pull/19647/commits/fe2da6a3b8babdeeb4e3cccfab710ddc8931f49a) - please let me know if there's anything else you think I should take a look at!

**Review by reiting — APPROVED** [review]

**Pull request merged**
